mithril-aggregator/src/snapshotter.rs

```diff
@@ -164,12 +170,21 @@ impl CompressedArchiveSnapshotter {
                 let mut tar = tar::Builder::new(enc);
 
                 tar.append_dir_all(".", &self.db_directory)
-                    .map_err(SnapshotError::CreateArchiveError)?;
+                    .map_err(SnapshotError::CreateArchiveError)
+                    .with_context(|| {
+                        format!(
+                            "GzEncoder Builder can not add directory: '{}' to the archive",
+                            self.db_directory.display()
+                        )
+                    })?;
 
                 let mut gz = tar
                     .into_inner()
-                    .map_err(SnapshotError::CreateArchiveError)?;
-                gz.try_finish().map_err(SnapshotError::CreateArchiveError)?;
+                    .map_err(SnapshotError::CreateArchiveError)
+                    .with_context(|| "GzEncoder can not unwrap the archive")?;
```

Comment:
I do not understand this error message. The [into_inner](https://docs.rs/tar/latest/tar/struct.Builder.html#method.into_inner) method seems to write everything on disk and raise any I/O error that could arise.